### PR TITLE
feat: Localize timestamps and durations

### PR DESCRIPTION
**Pull request by Aeledfyr**
This switches timestamp/duration handling to the new [Temporal API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Temporal).  This uses [fullcalendar/temporal-polyfill](https://github.com/fullcalendar/temporal-polyfill) as a polyfill for older browsers.

I've finished the clean-up; if this PR is too large to review, I can split it into a few logical sub-PRs (refactors of the existing usages, then piecemeal replacing with Temporal).  It may be less work to change it in one go, though.

## Screenshots

There are a bunch of places this could show up (but hopefully doesn't).  For now, I've made a test page at [Aeledfyr:nerimity-web/temporal-tests](https://github.com/Aeledfyr/nerimity-web/tree/temporal-tests) to test for edge cases.  This shows the result of each of the public duration-related functions:

## Did you test your code?

Yes, on Firefox & Chrome on Linux, but it probably needs more testing on other platforms.

## Checklist
- [ ] Changes are clear, concise, and easy to review  
- [x] Code has been tested and works as intended  
- [x] Text/content changes support internationalization (i18n)  
- [x] Any new user-facing strings are properly localized

## Summary by CodeRabbit

* **New Features**
  * Unified, locale-aware date/time formatting powered by a Temporal polyfill
  * New digital and compact duration formats and relative timestamp variants

* **Bug Fixes / Improvements**
  * More robust timezone handling with clearer fallback messaging
  * Added/updated locale strings for relative/datetime outputs

* **Style**
  * Numeric timestamps use tabular numbers for aligned display

### CODE REVIEW

File: src/common/date.ts [hunk 1]
```diff
@@ -1,131 +1,223 @@
...
+    const formatter = seconds ? timeFormatSeconds : dateFormat;
+    return formatter.format(timestamp.toPlainTime());
+  } else if (date.equals(yesterday.toPlainDate())) {
+    return t("datetime.yesterdayTime", { time: dateFormat.format(timestamp.toPlainTime()) });
+  } else {
+    return t("datetime.dateTime", {
+      date: dateFormat.format(timestamp.toPlainDate()),
+      time: dateFormat.format(timestamp.toPlainTime()),
+    });
+  }
```
[the remark this answers is not in the document]
Comment (Aeledfyr) [reply src/common/date.ts:163]: This is incorrect; [reference on MDN (though it isn't completely clear)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#:~:text=You%20can%20format%20different%20object%20types%20depending%20on%20which%20of%20the%20style%20shortcut%20options%20you%20include).  A `DateTimeFormatter` can format a `PlainTime` value if it has `timeStyle`, even if it also has `dateStyle`.  The same applies for `PlainDate`s, where formatting will work as long as there is a `dateStyle`.  This works on Chrome, Firefox, and fullcalendar/temporal-polyfill.

**Comment by SupertigerDev** [comment]
Unfortunatly this PR is too big yes, split it please. Also did you test if the polyfill actually works?
